### PR TITLE
Add test for Simple Payments button

### PR DESCRIPTION
**Pull request by Stojdza**
Simple Payment button test. It works against `wpcalypso.wordpress.com/gutenberg/post/`. 

To test: Make sure that test `Insert a payment button: @parallel` is passing, both desktop and mobile. `UnhandledPromiseRejectionWarning` should not appear.

Part of #1620

### CODE REVIEW

File: lib/gutenberg/gutenberg-editor-component.js [hunk 1]
```diff
@@ -248,6 +248,59 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
...
+		);
+		await driverHelper.setWhenSettable(
+			this.driver,
+			By.css( '.simple-payments__field-price .components-text-control__input' ),
+			price
+		);
+		if ( allowQuantity === true ) {
+			await driverHelper.clickWhenClickable(
+				this.driver,
+				By.css( '.components-form-toggle__input' )
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-component.js]: I think we should be a bit more specific on thits one - something like `.simple-payments__field-multiple .components-form-toggle__input`

File: lib/gutenberg/gutenberg-editor-component.js [hunk 2]
```diff
@@ -248,6 +248,59 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
 		await driverHelper.clickWhenClickable( this.driver, By.css( '.dashicons-no-alt' ) );
 	}
 
+	async insertPaymentButtonDetails( {
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-component.js]: Should this function be in its own block component - this file is getting pretty big - there's probably some other functions that should be split out also

File: lib/gutenberg/gutenberg-editor-component.js [hunk 3]
```diff
@@ -248,6 +248,59 @@ export default class GutenbergEditorComponent extends AsyncBaseContainer {
...
+		title = 'Button',
+		description = 'Description',
+		price = '1.00',
+		currency = 'AUD',
+		allowQuantity = true,
+		email = 'test@wordpress.com',
+	} = {} ) {
+		await driverHelper.isElementPresent(
```
Comment (alisterscott) [inline lib/gutenberg/gutenberg-editor-component.js]: I'm not sure what this is doing here - should it be waitUntilPresent?

File: lib/gutenberg/gutenberg-editor-component.js (hunk 2 above)
Comment (Stojdza) [reply lib/gutenberg/gutenberg-editor-component.js]: I made a separate component for payment block - `payment-block-component.js` and added requested changes there.

**Comment by Stojdza** [comment]
While executing test locally sometimes I was seeing `UnhandledPromiseRejectionWarning`, during handling multiple windows - [this](https://github.com/Automattic/wp-e2e-tests/pull/1661/commits/54e146719eb5907aaf8bf5a7fbbb2cc4859520b9) commit ensures that it doesn't happen anymore.

**Comment by Stojdza** [comment]
Since https://github.com/Automattic/wp-e2e-tests/pull/1660 is merged, I added `Insert a payment button into a page:` to this PR.

**Pull request merged**